hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -161,6 +167,109 @@ private static Node parseRsql(final String rsql) {
         }
     }
 
+
+    private static final class ValidationRSQLVisitor<A extends Enum<A> & FieldNameProvider> implements RSQLVisitor<Boolean, String>{
```

Comment:
As you've already mentioned just duplicating code is not a great idea (DRY principle, etc.). So my suggestion would be to either create and AbstractFieldNameRSQLVisitor class that exposes helper methods such as #getFieldEnumByName, #getSubAttributesFrom, etc. or alternatively use the composition by injecting some helper class to whom you could delegate the corresponding parse/validate operations

Reply:
I've created an abstract superclass where I moved all commonly used code by both RSQL visitors. The visitors now extend that abstract class.